multicluster/controllers/multicluster/node_controller.go

```diff
@@ -58,10 +62,11 @@ func NewNodeReconciler(
 		precedence = mcsv1alpha1.PrecedenceInternal
 	}
 	reconciler := &NodeReconciler{
-		Client:     client,
-		Scheme:     scheme,
-		namespace:  namespace,
-		precedence: precedence,
+		Client:            client,
+		Scheme:            scheme,
+		namespace:         namespace,
+		precedence:        precedence,
+		gatewayCandidates: make(map[string]string),
```

Comment:
I think usually map[string][boolean] is used for set.

Reply:
Done